infra/_modules/cms_function_app/monitoring.tf

```diff
@@ -0,0 +1,206 @@
+        time_window              = "PT7M"
+        time_aggregation         = "Average"
+        operator                 = "LessThan"
+        threshold                = 2000
+        divide_by_instance_count = false
+      }
+
+      scale_action {
+        direction = "Decrease"
+        type      = "ChangeCount"
+        value     = "1"
+        cooldown  = "PT5M"
+      }
+    }
+
+    rule {
+      metric_trigger {
+        metric_name              = "CpuPercentage"
+        metric_resource_id       = module.cms_fn.function_app.plan.id
+        metric_namespace         = "microsoft.web/serverfarms"
+        time_grain               = "PT1M"
+        statistic                = "Average"
+        time_window              = "PT7M"
+        time_aggregation         = "Average"
+        operator                 = "LessThan"
+        threshold                = 30
+        divide_by_instance_count = false
+      }
+
+      scale_action {
+        direction = "Decrease"
+        type      = "ChangeCount"
+        value     = "1"
+        cooldown  = "PT5M"
+      }
+    }
+
+    rule {
+      metric_trigger {
+        metric_name        = "MemoryPercentage"
+        metric_resource_id = module.cms_fn.function_app.plan.id
+        metric_namespace   = "microsoft.web/serverfarms"
+        time_grain         = "PT1M"
+        statistic          = "Average"
+        time_window        = "PT7M"
+        time_aggregation   = "Average"
+        operator           = "LessThan"
+        threshold          = 30
+      }
+      scale_action {
+        direction = "Decrease"
+        type      = "ChangeCount"
+        value     = "1"
+        cooldown  = "PT5M"
+      }
+    }
+  }
+
+  tags = var.tags
+}
+
+##########
+# Alerts #
+##########
+
+resource "azurerm_monitor_metric_alert" "cms_fn_health_check" {
+  name                = "${module.cms_fn.function_app.function_app.name}-health-check-failed"
+  resource_group_name = module.cms_fn.function_app.resource_group_name
+  scopes              = [module.cms_fn.function_app.function_app.id]
+  description         = "${module.cms_fn.function_app.function_app.name} health check failed"
+  severity            = 1
+  frequency           = "PT5M"
+  auto_mitigate       = false
+
+  criteria {
+    metric_namespace = "Microsoft.Web/sites"
+    metric_name      = "HealthCheckStatus"
+    aggregation      = "Average"
+    operator         = "LessThan"
+    threshold        = 50
+  }
+
+  action {
+    action_group_id = data.azurerm_monitor_action_group.error_action_group.id
+  }
+
+  tags = var.tags
+}
+
+resource "azurerm_monitor_metric_alert" "cms_fn_st_low_availability" {
+  name                = "[${module.cms_fn.storage_account.name}] Low Availability"
+  resource_group_name = module.cms_fn.function_app.resource_group_name
+  scopes              = [module.cms_fn.storage_account.id]
+  description         = "The average availability is less than 99.8%. Runbook: not needed."
+  severity            = 0
```

Comment:
**question:** do these two alarms (`cms_fn_st_low_availability` and `cms_fn_health_check`) have the same "importance"?

If so, I think we should assign the same severity to both (figure out whether `critical` or `error`).

Reply:
This apply also to: https://github.com/pagopa/io-services-cms/pull/885/files